crates/matrix-sdk-appservice/src/lib.rs

```diff
@@ -248,7 +264,15 @@ impl AppService {
     /// [assert the identity]: https://matrix.org/docs/spec/application_service/r0.1.2#identity-assertion
     pub async fn virtual_user(&self, localpart: Option<&str>) -> Result<Client> {
         let localpart = localpart.unwrap_or_else(|| self.registration.sender_localpart.as_ref());
-        self.virtual_user_builder(localpart).build().await
+        match self.default_request_config {
+            Some(config) => {
+                self.virtual_user_builder(localpart)
+                    .client_builder(Client::builder().request_config(config))
+                    .build()
+                    .await
+            }
+            None => self.virtual_user_builder(localpart).build().await,
+        }
```

Comment:
This can be simplified a bit by moving the `.build().await` outside the match (preferably by introducing a `let builder = match`).

Reply:
Good point, done. Sorry about all the extra pushes, I messed up the rebase a little. Should be fixed now though.